agent/csi/plugin.go

```diff
@@ -33,12 +50,133 @@ type NodePlugin struct {
 
 	// NodeClient is the node service client
 	NodeClient csi.NodeClient
+
+	// VolumeMap is the map from volume ID to Volume. Will place a volume once it is staged,
+	// remove it from the map for unstage.
+	// TODO: Make this map persistent if the swarm node goes down
+	VolumeMap map[string]*volumePublishStatus
+
+	// Lock for VolumeMap
+	Lock sync.RWMutex
+}
```

Comment:
Change all of these fields to be unexported; that is, change them to start with a lowercase letter.